src/lib/models/aic.rs

```diff
@@ -0,0 +1,186 @@
+        query_as!(
+            AIC,
+            r#"UPDATE aic
+            SET
+                cj_event_value = $1,
+                flow_id = $2,
+                created = $3,
+                expires = $4
+            WHERE id = $5
+			RETURNING *"#,
+            cj_event_value,
+            flow_id,
+            created,
+            expires,
+            id,
+        )
+        .fetch_one(self.db_pool)
+        .await
+    }
+
+    pub async fn fetch_one(&self) -> Result<AIC, Error> {
+        query_as!(AIC, "SELECT * FROM aic")
+            .fetch_one(self.db_pool)
+            .await
+    }
+
+    pub async fn fetch_one_by_id(&self, id: &Uuid) -> Result<AIC, Error> {
+        query_as!(AIC, "SELECT * FROM aic WHERE id = $1", id)
+            .fetch_one(self.db_pool)
+            .await
+    }
+
+    pub async fn fetch_one_by_flow_id(&self, flow_id: &str) -> Result<AIC, Error> {
+        query_as!(AIC, "SELECT * FROM aic WHERE flow_id = $1", flow_id)
+            .fetch_one(self.db_pool)
+            .await
+    }
+
+    pub async fn fetch_one_by_id_from_archive(&self, id: &Uuid) -> Result<AIC, Error> {
+        query_as!(AIC, "SELECT * FROM aic_archive WHERE id = $1", id)
+            .fetch_one(self.db_pool)
+            .await
+    }
+
+    pub async fn fetch_one_by_flow_id_from_archive(&self, flow_id: &str) -> Result<AIC, Error> {
+        query_as!(AIC, "SELECT * FROM aic_archive WHERE flow_id = $1", flow_id)
+            .fetch_one(self.db_pool)
+            .await
+    }
+
+    pub async fn create_archive_from_aic(&self, aic: &AIC) -> Result<AIC, Error> {
+        query_as!(
+            AIC,
+            r#"INSERT INTO aic_archive (id, cj_event_value, flow_id, created, expires)
+			VALUES ($1, $2, $3, $4, $5)
+			RETURNING *"#,
+            aic.id,
+            aic.cj_event_value,
+            aic.flow_id,
+            aic.created,
+            aic.expires
+        )
+        .fetch_one(self.db_pool)
+        .await
+    }
+
+    async fn create_archive_delete_aic(
+        &self,
+        transaction: &mut sqlx::Transaction<'_, sqlx::Postgres>,
+        aic: &AIC,
+    ) -> Result<(), Box<dyn std::error::Error>> {
+        query!("DELETE FROM aic WHERE id = $1", aic.id)
+            .execute(&mut *transaction)
+            .await?;
+        query!(
+            r#"INSERT INTO aic_archive (id, cj_event_value, flow_id, created, expires)
+			VALUES ($1, $2, $3, $4, $5)
+			RETURNING *"#,
+            aic.id,
+            aic.cj_event_value,
+            aic.flow_id,
+            aic.created,
+            aic.expires
+        )
+        .fetch_one(&mut *transaction)
+        .await?;
+        Ok(())
+    }
+
+    pub async fn archive_aic(&self, aic: &AIC) -> Result<(), Box<dyn std::error::Error>> {
+        // Wrap creating archive row and deleting aic row into one transaction
+        let mut transaction = self.db_pool.begin().await?;
+        self.create_archive_delete_aic(&mut transaction, aic)
```

Comment:
It doesn't look like you're using this method elsewhere. IMO dividing this into two methods seems superfluous.

Reply:
this was how the docs on transactions were laid out and so i assumed it was necessary for transactions to work.

Reply:
i'm going to leave this is as is for now. we can come back to it later.